packages/sharing-smart-contract/scripts/upgrade.js

```diff
@@ -1,16 +1,24 @@
 /* eslint-disable no-console */
-import { getEnvironment } from '@iexec/dataprotector-environments';
 import hre from 'hardhat';
-import { DATASET_REGISTRY_ADDRESS, POCO_ADDRESS } from '../config/config.js';
+import {
+  DATASET_REGISTRY_ADDRESS as defaultDatasetRegistryAddress,
+  POCO_ADDRESS as defaultPocoAddress,
+} from '../config/config.js';
 import { VOUCHER_HUB_ADDRESS } from '../test/bellecour-fork/voucher-config.js';
+import { getLoadFromEnv } from './singleFunction/utils.js';
 
 const { ethers, upgrades } = hre;
 
 async function main() {
   const { ENV } = process.env;
-  console.log(`Using ENV: ${ENV}`);
+  const loadFromEnv = getLoadFromEnv(ENV);
 
-  const { dataprotectorSharingContractAddress, addOnlyAppWhitelistRegistryContractAddress } = getEnvironment(ENV);
+  const {
+    DATAPROTECTOR_SHARING_ADDRESS = loadFromEnv('dataprotectorSharingContractAddress'),
+    ADD_ONLY_APP_WHITELIST_REGISTRY_ADDRESS = loadFromEnv('addOnlyAppWhitelistRegistryContractAddress'),
+    POCO_ADDRESS = defaultPocoAddress,
+    DATASET_REGISTRY_ADDRESS = defaultDatasetRegistryAddress,
```

Comment:
why not load POCO_ADDRESS & DATASET_REGISTRY_ADDRESS from env also ? As those addresses could be different for bubble ?

Reply:
Same, bubble will not be added to known environments as it constantly changes over redeployments and can be moved or shut down anytime, bubble should be treated as a custom ephemeral environment.
PoCo and DatasetRegistry address are not part of the environment package, they could be added but this is not needed for now because every environment is based on bellecour with a single instance of core contracts.
NB: this could change if staging moves on a dedicated chain or a dedicated instance of PoCo, but this is not something I have hands-on.